compiler/src/dotty/tools/repl/JLineTerminal.scala

```diff
@@ -18,7 +18,6 @@ final class JLineTerminal extends java.io.Closeable {
   // Logger.getLogger("org.jline").setLevel(Level.FINEST)
 
   private val terminal = TerminalBuilder.builder()
-    .dumb(false) // fail early if not able to create a terminal
```

Comment:
I am against enabling this by default. Last time I tried, the dumb terminal was very limited and I don't want to expose it to users. E.g.
- no syntax highlighting 
- no history
- no auto-completion
- no multi-line editing

You can experiment with dumb terminal with these changes:
```diff
diff --git a/compiler/src/dotty/tools/repl/JLineTerminal.scala b/compiler/src/dotty/tools/repl/JLineTerminal.scala
index 22e44e4a5..9c33a7b87 100644
--- a/compiler/src/dotty/tools/repl/JLineTerminal.scala
+++ b/compiler/src/dotty/tools/repl/JLineTerminal.scala
@@ -17,9 +17,12 @@ final class JLineTerminal extends java.io.Closeable {
   // import java.util.logging.{Logger, Level}
   // Logger.getLogger("org.jline").setLevel(Level.FINEST)
 
-  private val terminal = TerminalBuilder.builder()
-    .dumb(false) // fail early if not able to create a terminal
-    .build()
+  // private val terminal = TerminalBuilder.builder()
+  //   .dumb(false) // fail early if not able to create a terminal
+  //   .build()
+  import java.lang.System
+  private val terminal =
+    new org.jline.terminal.impl.DumbTerminal(System.in, System.out)
   private val history = new DefaultHistory
 
   private def blue(str: String) = Console.BLUE + str + Console.RESET
```

I would be Ok to make this configurable in order to use dumb terminal in tests. However, we clearly need to document what can be tested